src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/IProjectSystemCache.cs

```diff
@@ -107,5 +121,12 @@ public interface IProjectSystemCache
         /// Clears all project cache data.
         /// </summary>
         void Clear();
+
+        /// <summary>
+        /// Reset the dirty flag to 0 (is Not Dirty) if the flag was already set.
+        /// This is public so that external callers can inform the cache that they have consumed the updated cache event.
+        /// </summary>
+        /// <returns><code>true</code> if the cache was dirty before and <code>false</code> otherwise</returns>
+        bool TestResetDirtyFlag();
```

Comment:
Because the method resets the DirtyFlag only if the flag was set before hence Test Prefix. it behaves like a Test-Set Lock/TSL.